openlibrary/macros/LoanStatus.html

```diff
@@ -15,6 +15,20 @@
 $ ocaid = doc.get('ocaid') or availability.get('identifier')
 $ work_key = work_key or (doc.get('works') and doc.works[0].key)
 
+$ book_provider = 'ia'
+
+$# Check for Project Gutenberg ID
+$ gutenberg_ids = doc.get('identifiers', {}).get('project_gutenberg') or doc.get('id_project_gutenberg')
+$ gutenberg_id = gutenberg_ids and gutenberg_ids[0]
+$if (gutenberg_id and (not ocaid or ocaid.endswith('gut'))):
+  $ book_provider = 'gutenberg'
+
+$# Check for LibriVox ID
+$ librivox_ids = doc.get('identifiers', {}).get('librivox') or doc.get('id_librivox')
+$ librivox_id = librivox_ids and librivox_ids[0]
+$if librivox_id:
+  $ book_provider = 'librivox'
```

Comment:
Would it make sense to do all partner services in a single check to prevent LoanStatus from ballooning in scope?

e.g.  would it make sense to check if the ID is in an approved list of providers? (e.g. `in ["gutenberg", "standardebooks", "librivox"]`) and then call the right template accordingly, like we're doing below

Reply:
I think this is where some DRY-ing would be potentially useful; basically having an array of BookProviders, where each provider has its own logic in the python, and manages its own templates. But there were a few odd edge cases, so opted to leave it like this for now.